diff --git a/ios/ListView.swift b/ios/ListView.swift
index a7ca51df0921b2ee87d1c32be6ed409c8a2f0d0d..f637163c58da99d050e2eb30640d06a19f444a99 100644
--- a/ios/ListView.swift
+++ b/ios/ListView.swift
@@ -5,6 +5,8 @@ import SwiftUI

 final class ListProps: UIBaseViewProps {
   @Field var selection: [Either<String, Double>]?
+  @Field var hideScrollIndicators: Bool = true
+
   var onSelectionChange = EventDispatcher()
 }

@@ -13,7 +15,7 @@ struct ListView: ExpoSwiftUI.View {
   @State private var selection = Set<AnyHashable>()

   var body: some View {
-    List(selection: $selection) {
+    let list = List(selection: $selection) {
       Children()
     }
     .onAppear {
@@ -25,6 +27,12 @@ struct ListView: ExpoSwiftUI.View {
     .onChange(of: selection) { newSelection in
       handleSelectionChange(selection: newSelection)
     }
+
+    if #available(iOS 16.0, *) {
+      list.scrollIndicators(props.hideScrollIndicators ? .hidden : .visible)
+    } else {
+      list
+    }
   }

   func handleSelectionChange(selection: Set<AnyHashable>) {
